go/stream/message.go

```diff
@@ -56,6 +58,34 @@ type MessageProperties struct {
 	EncryptionKeyID string `json:"encryptionKeyID,omitempty"` // optional
 }
 
+func (m MessageProperties) LoggerFields() []logger.Field {
+	mapProperties := ToMapProperties(m)
+	keysToLog := []string{
```

Comment:
@mihir20 

> Try to write a method that is as lean as possible

You're still doing extra allocations because you create a map that you don't really need by calling `ToMapProperties`.

https://github.com/rudderlabs/rudder-schemas/pull/41